sys/x86/x86/mca.c

```diff
@@ -426,7 +430,12 @@ mca_log(const struct mca_record *rec)
 	uint16_t mca_error;
 
 	if (mca_mute(rec))
-	    	return;
+		return;
+
+	if (!log_corrected && (rec->mr_status & MC_STATUS_UC) == 0 &&
+	    (!tes_supported(rec->mr_mcg_cap) ||
+	    ((rec->mr_status & MC_STATUS_TES_STATUS) >> 53) != 0x2))
```

Comment:
This is pretty gnarly looking, but at least it's clear enough if you keep looking a few lines down.